### PR TITLE
A new encoding for the CoMID flags type

### LINKED ISSUE #174: Change CoMID flags to align with DICE errata
Change CoMID flags to align with DICE errata, i.e., we also need the dual for each flag value.

This means we now need three states for each state: asserted, negated, unknown, which calls for a different shape of the flags type:

```
flags = {
	? comid.debug => bool
	? comid.configured => bool
	; ...
}
```

### PR DESCRIPTION
**Pull request by thomas-fossati**
Fixes #174

### CODE REVIEW

**Review by nedmsmith — APPROVED** [review]

**Review by yogeshbdeshpande — APPROVED** [review]

File: concise-mid-tag.cddl [hunk 1]
```diff
@@ -174,28 +174,17 @@ tagged-svn = #6.552(svn)
 tagged-min-svn = #6.553(min-svn)
 svn-type-choice = tagged-svn / tagged-min-svn
 
-flags-type = [
```
Comment (henkbirkholz) [inline concise-mid-tag.cddl:177]: So we are moving from bitmask to map with labels ("code points") here. Is this a general trend or why would we be doing that only here?

Comment (thomas-fossati) [reply concise-mid-tag.cddl:177]: > So we are moving from bitmask to map with labels ("code points") here. Is this a general trend or why would we be doing that only here?

I don't see any other places in the CDDL where we use bit-masks (including imported modules, i.e., CoSWID):

```
╭─thofos01 at tho in ~/Documents/github.com/ietf-rats/ietf-corim-cddl on new-flags-encoding✘✘✘
╰─± grep -c '.bits' *.cddl
comid-autogen.cddl:0
comid-code-points.cddl:0
common.cddl:0
concise-mid-tag.cddl:0
concise-rim.cddl:0
concise-swid-tag.cddl:0
corim-autogen.cddl:0
corim-code-points.cddl:0
corim.cddl:0
flags.cddl:0
generic-non-empty.cddl:0
signed-corim.cddl:0
xcorim-autogen.cddl:0
xcorim-code-points.cddl:0
xcorim.cddl:0
```

So, apparently `flags-type` was the exception we got rid of :-)

Comment (henkbirkholz) [reply concise-mid-tag.cddl:177]: oh. maybe I should grep before I ask for an obvious answer. Then I agree with this change!

**Review by henkbirkholz — APPROVED** [review]

File: concise-mid-tag.cddl [hunk 2]
```diff
@@ -149,7 +149,7 @@ measurement-values-map = non-empty<{
   ? comid.ver => version-map
   ? comid.svn => svn-type-choice
   ? comid.digests => digests-type
-  ? comid.flags => flags-type
+  ? comid.flags => flags-map
```
[the remark this answers is not in the document]
Comment (thomas-fossati) [reply concise-mid-tag.cddl:152]: (resolving as per offline conversation between @yogeshbdeshpande and @thomas-fossati)

File: concise-mid-tag.cddl [hunk 3]
```diff
@@ -174,12 +174,8 @@ tagged-svn = #6.552(svn)
 tagged-min-svn = #6.553(min-svn)
 svn-type-choice = tagged-svn / tagged-min-svn
 
-flags-type = [
-  flags : bytes .bits $operational-flags
-  ? mask : bytes .bits $operational-flags
-]
 
-; operational-flags maps DiceTcbInfo.flags, i.e.:
+; flags-map maps DiceTcbInfo.flags, i.e.:
```
[the remark this answers is not in the document]
Comment (thomas-fossati) [reply concise-mid-tag.cddl]: Done in 7841590

File: concise-mid-tag.cddl [hunk 4]
```diff
@@ -190,12 +186,16 @@ flags-type = [
...
 ;
-$operational-flags /= &( not-configured: 0 )
-$operational-flags /= &( not-secure: 1 )
-$operational-flags /= &( recovery: 2 )
-$operational-flags /= &( debug: 3 )
-$operational-flags /= &( not-replay-protected: 4 )
-$operational-flags /= &( not-integrity-protected: 5 )
+; plus all the contraindicated / dual states
+flags-map = {
+  ? comid.operational-flag-configured => bool
```
[the remark this answers is not in the document]
Comment (thomas-fossati) [reply concise-mid-tag.cddl:180]: (resolving as per offline conversation between @yogeshbdeshpande and @thomas-fossati)

File: concise-mid-tag.cddl [hunk 5]
```diff
@@ -174,28 +174,17 @@ tagged-svn = #6.552(svn)
...
-$operational-flags /= &( not-integrity-protected: 5 )
+; flags-map maps DiceTcbInfo.flags
+flags-map = {
+  ? comid.operational-flag-configured => bool
+  ? comid.operational-flag-secure => bool
+  ? comid.operational-flag-recovery => bool
+  ? comid.operational-flag-debug => bool
+  ? comid.operational-flag-replay-protected => bool
+  ? comid.operational-flag-integrity-protected => bool
+  * $$flags-map-extension
```
Comment (henkbirkholz) [inline concise-mid-tag.cddl:186]: also, extensibility ftw

**Pull request merged**